internal/controller/controller.go

```diff
@@ -46,14 +48,12 @@ func (c *Controller) Start(ctx context.Context) error {
 		return fmt.Errorf("failed to probe tracepoint: %w", err)
 	}
 
-	presentCores, err := numcpus.GetPresent()
+	traceHandlerCacheSize, err :=
```

Comment:
IMO, we can move this into the tracehandler. `reporter.CacheSize` is a different value and should be determined inside the reporter (or optionally make it configurable).
We can do so in a separate PR, so that this PR becomes smaller and more focused.

Reply:
See https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/215